app/models/work_package.rb

```diff
@@ -382,8 +382,18 @@ def update_done_ratio_from_status
   # check if user is allowed to edit WorkPackage Journals.
   # see Acts::Journalized::Permissions#journal_editable_by
   def journal_editable_by?(journal, user)
-    user.allowed_in_project?(:edit_work_package_notes, project) ||
-      (user.allowed_in_work_package?(:edit_own_work_package_notes, self) && journal.user_id == user.id)
+    can_edit_journal_note = ->(edit_others_notes_permission, edit_own_notes_permission) {
+      user.allowed_in_project?(edit_others_notes_permission, project) ||
+        (user.allowed_in_work_package?(edit_own_notes_permission, self) && journal.user_id == user.id)
+    }
```

Comment:
This I would also extract into a simple method that is then called.

Reply:
This also has both calls in it. The one on the project and the one on the work package level.